lib/ansible/module_utils/acme.py

```diff
@@ -654,10 +656,21 @@ def get_request(self, uri, parse_json_result=True, headers=None, get_only=False,
         else:
             result = content
 
-        if fail_on_error and info['status'] >= 400:
+        if fail_on_error and (info['status'] < 200 or info['status'] >= 400):
             raise ModuleFailException("ACME request failed: CODE: {0} RESULT: {1}".format(info['status'], result))
         return result, info
 
+    def _assert_fetch_url_success(response, info, allow_redirect=False, allow_client_error=True, allow_server_error=True):
```

Comment:
This should be at top level, and not be a class method. (Here, `response` will be `self`.)